app/services/nomis/client.rb

```diff
@@ -13,7 +13,11 @@ def initialize(root, log404 = true)
       @connection = Faraday.new do |faraday|
         faraday.request :retry, max: 3, interval: 0.05,
                                 interval_randomness: 0.5, backoff_factor: 2,
-                                exceptions: [Faraday::ClientError, 'Timeout::Error']
+                                #  this looks like a strange list - client errors are unlikely to go away
+                                exceptions: [Faraday::ClientError, 'Timeout::Error'],
+                                # we seem to get 502 and 504 statuses from a gateway this side of
+                                # the Prison API - so retry if we get one of those.
+                                retry_statuses: [502, 504]
```

Comment:
Would it make sense to go with something more like this?

```suggestion
                                exceptions: [
                                  Faraday::ClientError, # retry on 4xx errors
                                  Faraday::ServerError, # retry on 5xx errors
                                  Errno::ETIMEDOUT, Timeout::Error, Faraday::TimeoutError # retry on network timeouts
                                ],
```

i.e.
- retry 4xx responses
- retry 5xx responses (which covers 502 and 504)
- the final 3 exception types (`Errno::ETIMEDOUT`, `Timeout::Error`, `Faraday::TimeoutError`) mean that we retry on network-related exceptions, as per [the default config for retries](https://lostisland.github.io/faraday/middleware/retry#specify-which-exceptions-should-trigger-a-retry:~:text=that%20will%20replace%20the%20default%20list,of%20network%2Drelated%20exceptions%3A%20Errno%3A%3AETIMEDOUT%2C%20Timeout%3A%3AError%2C%20Faraday%3A%3ATimeoutError) which is otherwise overridden when we set this option.

Reply:
I'm not sure we want to retry on all 500-type responses, just those that we know come from the gateway between us and the Prison API. I've added another commit so that we only add the 502 and 504 statuses, and most other stuff is defaulted.